hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/weighted/ConsumerNodeLoadDecoder.java

```diff
@@ -0,0 +1,58 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.weighted;
+
+import org.agrona.concurrent.UnsafeBuffer;
+import org.slf4j.Logger;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.consumers.subscription.id.SubscriptionId;
+import pl.allegro.tech.hermes.consumers.subscription.id.SubscriptionIds;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.sbe.stubs.ConsumerLoadDecoder;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.sbe.stubs.ConsumerLoadDecoder.SubscriptionsDecoder;
+import pl.allegro.tech.hermes.consumers.supervisor.workload.sbe.stubs.MessageHeaderDecoder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+class ConsumerNodeLoadDecoder {
+
+    private static final Logger logger = getLogger(ConsumerNodeLoadDecoder.class);
+
+    private final SubscriptionIds subscriptionIds;
+
+    ConsumerNodeLoadDecoder(SubscriptionIds subscriptionIds) {
+        this.subscriptionIds = subscriptionIds;
+    }
+
+    ConsumerNodeLoad decode(byte[] bytes) {
```

Comment:
What do You think about splitting this function?

Reply:
Done.